wire-ios-request-strategy/Sources/Payloads/Processing/Helpers/MLSEventProcessor.swift

```diff
@@ -51,84 +76,176 @@ final class MLSEventProcessor: MLSEventProcessing {
+        in context: NSManagedObjectContext
+    ) async {
+        WireLogger.mls.info("MLS event processor is processing welcome message")
 
-        let mlsService = await context.perform { context.mlsService }
-        guard let mlsService else {
+        guard let mlsService = await context.perform({ context.mlsService }) else {
             return logWarn(aborting: .processingWelcome, withReason: .missingMLSService)
         }
 
+        let oneOnOneResolver = OneOnOneResolver(
+            protocolSelector: OneOnOneProtocolSelector(),
+            migrator: OneOnOneMigrator(mlsService: mlsService)
+        )
+
+        await process(
+            welcomeMessage: welcomeMessage,
+            conversationID: conversationID,
+            in: context,
+            mlsService: mlsService,
+            oneOnOneResolver: oneOnOneResolver
+        )
+    }
+
+    func process(
+        welcomeMessage: String,
+        conversationID: QualifiedID,
+        in context: NSManagedObjectContext,
+        mlsService: MLSServiceInterface,
+        oneOnOneResolver: OneOnOneResolverInterface
+    ) async {
         do {
             let groupID = try await mlsService.processWelcomeMessage(welcomeMessage: welcomeMessage)
-
-            await context.perform {
-
-                guard let conversation = ZMConversation.fetch(with: groupID, in: context) else {
-                    return self.logWarn(aborting: .processingWelcome, withReason: .other(reason: "conversation does not exist in db"))
+            await mlsService.uploadKeyPackagesIfNeeded()
+            await conversationService.syncConversation(qualifiedID: conversationID)
+
+            let conversation: ZMConversation? = await context.perform {
+                guard let conversation = ZMConversation.fetch(
+                    with: conversationID,
+                    in: context
+                ) else {
+                    return nil
                 }
+
+                conversation.mlsGroupID = groupID
                 conversation.mlsStatus = .ready
-                context.saveOrRollback()
 
-                Logging.mls.info(
-                    "MLS event processor set mlsStatus to (\(String(describing: conversation.mlsStatus)) for group (\(groupID.safeForLoggingDescription))"
-                )
+                return conversation
+            }
+
+            guard let conversation else { return }
+
+            await resolveOneOnOneConversationIfNeeded(
+                conversation: conversation,
+                mlsService: mlsService,
+                oneOneOneResolver: oneOnOneResolver,
+                in: context
+            )
+
+        } catch {
+            WireLogger.mls.warn("MLS event processor aborting processing welcome message: \(String(describing: error))")
+            return
+        }
+    }
+
+    private func resolveOneOnOneConversationIfNeeded(
+        conversation: ZMConversation,
+        mlsService: MLSServiceInterface,
```

Comment:
`mlsService` is not used

Reply:
https://github.com/wireapp/wire-ios/pull/740/commits/7e2a3df4226ecd05d1235ad790be2011e3a035f4